src/opi/input/structures/properties.py

```diff
@@ -0,0 +1,321 @@
+        """
+        # > Select mode
+        mode_functions = {
+            "goat": cls.goat_energies,
+            "docker": cls.docker_energies,
+        }
+
+        comments_tuple: tuple[str, ...] | None = None
+
+        # > Convert comments to tuple
+        if isinstance(comment_symbols, str):
+            comments_tuple = (comment_symbols,)
+        elif isinstance(comment_symbols, Sequence):
+            comments_tuple = tuple(comment_symbols)
+
+        # > Skip arbitrary number of empty and user comment lines at the beginning
+        while (line := xyz_lines.readline()) != "":
+            if not line.lstrip():
+                continue
+            # > Check for comment line. Ignore empty/whitespace lines
+            elif comments_tuple and line.lstrip().startswith(comments_tuple):
+                continue
+            else:
+                break
+
+        # > No data available in the buffer
+        if line == "":
+            return None
+
+        # > Fetch number of atoms
+        try:
+            natoms = int(line.split()[0])
+        except (ValueError, IndexError) as err:
+            raise ValueError(
+                f"Line {xyz_lines.line_number}: Could not read number of atoms at the beginning of xyz data"
+            ) from err
+
+        # > Comment line
+        line = xyz_lines.readline()
+        if line == "":
+            raise ValueError(
+                f"Line {xyz_lines.line_number}: Comment line is not present in xyz data"
+            )
+
+        # > Analyse comment line
+        properties = mode_functions[mode](line)
+
+        # > Skip the remaining structure
+        for iline in range(natoms):
+            line = xyz_lines.readline()
+            # > empty lines are not allowed
+            if line == "":
+                raise ValueError(f"Line {xyz_lines.line_number}: Incomplete xyz file buffer")
+
+        return properties
+
+    @classmethod
+    def docker_energies(cls, line: str) -> "Properties":
+        """Function for reading docker energies from string and return a Properties object."""
+        numbers = [float(m) for m in re.findall(RGX_INT_AND_FLOAT, line)]
+        # > Parse the comment line
+        try:
+            # > ID of the structure
+            structure_id = int(numbers[0])
+            # > Total energy is second number (Eh)
+            energy_total = numbers[1]
+            # > Relative energy is the third number (kcal/mol)
+            energy_relative = numbers[2]
+        except IndexError:
+            raise ValueError("Could not parse docker energies from comment line.")
+        properties = Properties(
+            structure_id=structure_id, energy_total=energy_total, energy_relative=energy_relative
+        )
+        return properties
+
+    @classmethod
+    def goat_energies(cls, line: str) -> "Properties":
+        """Function for reading goat energies from string and return a Properties object."""
```

Comment:
1)

```suggestion
        """Function for reading GOAT energies from string and return a Properties object."""
```
2) From any string? Please document what the string has to look like.